src/resource/worker-control.c

```diff
@@ -531,7 +531,7 @@ static int
 				goto fail;
 			}
 
-			if (!is_worker && chan->spec->wire.ext.used && chan->spec->wire.ext.pipe.fd_redir >= 0) {
+			if (is_worker && chan->spec->wire.ext.used && chan->spec->wire.ext.pipe.fd_redir >= 0) {
```

Comment:
Yeah, right... At first I was thinking about a possibility to be able to redirect output on main process side ("proxy") through a pipe to worker so worker can do some further processing on the data stream the proxy would normally just write (e.g. when writing a file by main process, then redirecting that writing to worker so it can do further reformatting, decorating the data with certain format or sending the data to different locations, files...). Still, we'd need the redirection on the worker side. If we ever needed this, we can easily extend the code, but probably not needed now - the use case seems quite artificial for now. The use case with redirecting the "pipe read end" on worker side to external worker's input is surely much more common and makes more sense as you say...